backend/lambdas/process_post/lambda.py

```diff
@@ -0,0 +1,193 @@
+
+    bedrock_llm = ChatBedrock(
+        model_id=MODEL_ID,
+        model_kwargs=INFORMATION_EXTRACTION_MODEL_PARAMETERS,
+        client=bedrock_runtime,
+    )
+
+    claude_information_extraction_prompt_template = INFORMATION_EXTRACTION_PROMPT_SELECTOR.get_prompt(MODEL_ID)
+
+    print("The prompt template")
+    print(claude_information_extraction_prompt_template)
+
+    print(claude_information_extraction_prompt_template.format(
+        text=text,
+        sentiments=sentiments
+        )
+    )
+
+    structured_llm = bedrock_llm.with_structured_output(ExtractedInformation)
+
+    structured_chain = claude_information_extraction_prompt_template | structured_llm
+
+    information_extraction_obj = structured_chain.invoke({
+        "text": text,
+        "sentiments": sentiments
+    })
+
+    print("Information extraction object")
+    print(type(information_extraction_obj))
+    print(information_extraction_obj)
+
+    return information_extraction_obj
+
+@logger.inject_lambda_context(log_event=True)
+def handler(event, _context: LambdaContext):
+
+        item = event
+
+        logger.info('Item:')
+        logger.info(item)
+
+        #Attemp to categorize item
+        text = item['text']
+        logger.info(f'Text: {text}')
+
+        text = demoji.replace(text, "")
+        item['text_clean'] = text
+
+        try:
+
+            #meta_topics_str = ','.join(META_TOPICS)
+            topic_match = text_topic_match(META_TOPICS_STR, item['text_clean'])
+
+            if topic_match.is_match == True and len(topic_match.related_topics) > 0:
+
+                try:
+
+                    text_insight = text_information_extraction(META_SENTIMENTS_STR, item['text_clean'])
+                    logger.info(f'Text insights:')
+                    logger.info(text_insight)
+
+                    if text_insight is not None:
+                        item['process_post'] = True
+                        logger.info("Removing <UNKNOWN> values")
+                        logger.info(text_insight)
+                        unknown_to_empty_text_values(text_insight)
+                        logger.info("<UNKNOWN> removed")
+                        logger.info(text_insight)
+                        item.update(text_insight)
+                        item["meta_topics"] = topic_match.related_topics
+
+                        # Process location only if there is one
+                        if len(item['location']) > 0:
+                            item['process_location'] = True
+                        else:
+                            item['process_location'] = False
+
+                        item['model'] = MODEL_ID
+
+                        logger.info(f'Invoking next function')
+                        logger.info(item)
+                    else:
+                        item['process_post'] = False
+                        logger.info(f'Nothing else to do')
+                        logger.info(item)
+
+                    return item
```

Comment:
another minor: I understood the idea of returning an altered version of the input event, but this is slightly confusing.

Now that you've met Pydantic, how would you feel about typing the output of your handler, say, with a `ProcessedPost` type? It makes it clearer what this method outputs?

Reply:
Fixed in [e91f7a0](https://github.com/aws-samples/ai-powered-text-insights/pull/3/commits/e91f7a05ccb41059a49562d086e6824bc3af9ecc)